IMAP-Science-Operations-Center/sds-data-manager
# First pass at processing architecture

# Change Summary

## Overview
Creates code for part of this diagram: https://lasp.colorado.edu/galaxy/x/dJJaCg (starting with L1A).

## New Files
l1b_Codice.py - template lambda code (work in progress)
l1c_Codice.py - template lambda code (work in progress)

data_storage_stack.py - for data storage
networking_stack.py - for vpc and security groups
processing_stack.py - for processing

batch_compute_resources.py - reusable construct for batch processing
instrument_lambdas.py - construct for setting up lambdas
sdc_step_function.py - step function construct

## Deleted Files
None

## Updated Files
stackbuilder.py - where stacks are deployed
sds_data_manager_stack.py - modified docstring

## Testing
You can test using what I have deployed already to AWS:
- Look at the rules for EventBridge
- Put a file in the l1a_Codice s3 bucket: archive-lcs-dev
- Observe the step functions: l1b-Codice-lcs-dev-step-function, l1c-Codice-lcs-dev-step-function
- Observe that files are dropped into the directories l1b_Codice, l1c_Codice in archive-lcs-dev

Note: I created a dummy Docker image that is in the ECRs: l1b-codice-lcs-dev-repo and l1c-codice-lcs-dev-repo. You could also copy those into your own ECR and use your own AWS resources if you wanted to test that way.
